File: csharp/MapleStory.OpenAPI/Src/Dto/History/PotentialHistoryResponseDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 잠재능력 재설정 히스토리 응답 정보
    /// </summary>
    public class PotentialHistoryResponseDTO
    {
        /// <summary>
        /// 결과 건 수
        /// </summary>
        [JsonProperty("count")]
        public int Count;

        /// <summary>
        /// 잠재능력 재설정 히스토리
        /// </summary>
        [JsonProperty("potential_history")]
        public List<PotentialHistoryDTO> PotentialHistory;

        /// <summary>
        /// 페이징 처리를 위한 cursor
        /// </summary>
        [JsonProperty("next_cursor")]
        public string?  NextCursor;
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/kms/dto/history/PotentialHistoryResponseDTO.java
package dev.spiralmoon.maplestory.api.kms.dto.history;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;

/**
 * 잠재능력 재설정 히스토리 응답 정보
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class PotentialHistoryResponseDTO {

    /**
     * 결과 건 수
     */
    @JsonProperty("count")
    private int count;

    /**
     * 잠재능력 재설정 히스토리
     */
    @JsonProperty("potential_history")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<PotentialHistoryDTO> potentialHistory;

    /**
     * 페이징 처리를 위한 cursor
     */
    @JsonProperty("next_cursor")
    private String nextCursor;
}
